docs/expected_loss.ipynb

```diff
@@ -0,0 +1,203 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "prompt-victorian",
+   "metadata": {},
+   "source": [
+    "# Federations vs. simple risk splitting\n",
+    "\n",
+    "After a recent [Twitter discussion](https://twitter.com/liber_liver/status/1453654819227512834) I wanted to explore how much better federated custodians protect against malicious actors than naively splitting one's capital over multiple centralized custodians. We assume that each federation member/custodian has a probability $P_{mal}$ of being malicios.\n",
+    "\n",
+    "We are now interested in the expected loss $E(\\text{loss})$ a user suffers due to that. Loss is defined as losing access to their funds. In the federated setting that can be either due to theft or due to the adversary stopping the federation (which is easier). There exist countermeasures against adversaries merely stopping the federation (e.g. for ransom) but we will pessimistically assume that both failure modes mean loss of funds for the user (even if only for a limited time). In the centralized custodial case on the other hand the money gets stolen or not.\n",
+    "\n",
+    "We explore both federations of optimal sizes $(i \\cdot 3) + 1$ for as well as federations of suboptimal sizes. We see that adding members to a federation of already optimal size hurts resilience in our model since there are more potentially malicious parties but the threshold for how many malicious parties the system can handle does not change."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "prescribed-assumption",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def max_faulty_liveness(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for liveness\"\"\"\n",
+    "    return (n - 1) // 3\n",
+    "\n",
+    "def max_faulty_fund_safety(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for fund safety\"\"\"\n",
+    "    return n - max_faulty_liveness(n) - 1\n",
```

Comment:
A reference to the technical paper describing the Honey Badger consensus mechanism would help explain these thresholds.